akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ShardingDirectReplication.scala

```diff
@@ -66,6 +66,7 @@ private[akka] object ShardingDirectReplication {
 
       Behaviors.receiveMessage {
         case WrappedPublishedEvent(event) =>
+          // FIXME this doesn't support migration from non-replicated
```

Comment:
There can be more things. I haven't looked or tested thoroughly. Then of course in grpc replication as well.